epicsdbbuilder/const_array.py

```diff
@@ -0,0 +1,92 @@
+from decimal import Decimal
+from .recordbase import quote_string
+from .parameter import Parameter
+
+__all__ = ['ConstArray']
+
+
+class ConstArray:
+    """Constant Link Values. EPICS Base 3.16.1 and above.
+
+    Example: PY Source
+    ------------------
+    `r = records.lsi('r', INP=ConstArray(['Plain String not DBLINK']))`
+
+    Example: Generated DB
+    ---------------------
+    `field(INP, ["Plain String not DBLINK"])`
+    """
+
+    def __init__(self, value):
+        """Constructor.
+
+        Parameters
+        ----------
+        value : iterator
+            Homogeneous non-empty list of values.
+        """
+        self.value = value
+
+    @property
+    def value(self):
+        """A list of values.
+        """
+        return self._value
+
+    @value.setter
+    def value(self, value):
+        self._value = self._sanitize(value, 'ConstArray')
+
+    def _sanitize(self, raw_value, context_name):
+        try:
+            value_list = list(raw_value)
+        except TypeError:
```

Comment:
I understand and I agree. Yet, let me add my motivation.

If we leave it up to `list()`, the user will get `'int' object is not iterable` if one attempts to assign an integer instead of anything else. Unless the user knows internals of epicsdbbuilder, the user needs to read corresponding source files. Alternatively, the user will keep a list of error messages and their interprettation. At the same time, epicsdbbuilder knows potential cause of the situation. It can help or guide the user a little bit so that the user can focus on the task rather then on the tool in majority of cases.

Would you like me to remove this try-except?

Reply:
I think you have responded to my question in one of your later responses. I will update the code accordingly.